test/parallel/test-abortcontroller.js

```diff
@@ -231,6 +231,42 @@ const { setTimeout: sleep } = require('timers/promises');
   AbortSignal.timeout(1_200_000);
 }
 
+// Make sure AbortSignal.timeout not leaking memory
+{
+  function getMemoryAllocatedInMB() {
+    return Math.round(process.memoryUsage().rss / 1024 / 1024 * 100) / 100;
+  }
+
+  async function createALotOfAbortSignals() {
+    for (let i = 0; i < 10000; i++) {
+      function lis() {
+
+      }
+
+      const timeoutSignal = AbortSignal.timeout(1_000_000_000);
+      timeoutSignal.addEventListener('abort', lis);
+      aborted(timeoutSignal, {});
+      timeoutSignal.removeEventListener('abort', lis);
+    }
+
+    await sleep(10);
+    globalThis.gc();
+  }
+
+  (async () => {
+    // Making sure we create some data so we won't catch something that is related to the infra
+    await createALotOfAbortSignals();
+
+    const currentMemory = getMemoryAllocatedInMB();
+
+    await createALotOfAbortSignals();
+
+    const newMemory = getMemoryAllocatedInMB();
+
+    strictEqual(newMemory - currentMemory < 100, true, new Error(`After consuming 10 items the memory increased by ${Math.floor(newMemory - currentMemory)}MB`));
```

Comment:
Is there a better way to test 🤔 feels like this could be flakey

Reply:
This is a good question but to reduce flakiness this test would be to not run in parallel with other tests, will move it to a different file

Reply:
https://github.com/nodejs/node/pull/48952#discussion_r1278356019

Reply:
this 100 sounds like a magic number which would not be truly reliable... (I don't have a better idea of how to test this though)

Reply:
Removed the test as it did not work for some reason...

Reply:
@rluvaton I wrote this in another conversation, but I feel it belongs here...
Maybe using a `WeakRef[]` etc we can ensure the values are gone after `globalThis.gc();`, instead of checking an increase in memory? (meaning add them wrapped by `WeakRef` into an array during the `createALotOfAbortSignals` setup)

Reply:
great idea the test was added :)

Reply:
removed the test as it can be flaky and instead checked that the memory leak warning was not emitted...